commons/src/main/java/com/powsybl/commons/config/PlatformConfigNamedProvider.java

```diff
@@ -0,0 +1,197 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.commons.config;
+
+import com.google.common.collect.Lists;
+import com.powsybl.commons.PowsyblException;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.ServiceLoader;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * A provider that can be loaded by by Java's ServiceLoader based on its name
+ * present in an entry in the PlatformConfig.
+ *
+ * @author Jon Harper <jon.harper at rte-france.com>
+ */
+public interface PlatformConfigNamedProvider {
+
+    /**
+     * Get the name.
+     *
+     * @return the name
+     */
+    String getName();
+
+    /**
+     * Get the Provider name used for identifying this provider in the
+     * PlatformConfig. Defaults to getName(). Override this method only getName() is
+     * already implemented and returns the wrong name.
+     *
+     * @return the name
+     */
+    default String getPlatformConfigName() {
```

Comment:
Not sure I understand how this method is different from `getName()`? It is never overriden in the code, isn't it? In which case would it be overriden?

Reply:
This is done to automatically reuse the implement of "getName" methods from other interfaces (e.g. Versionable ) in most cases, but still allow to have different names in case you need it. It is not currently needed in our code now.

Reply:
ok